### PR TITLE
feat: add support for flink application version

### PR DESCRIPTION
**Pull request by ivan-savciuc**
## About this change—what it does

Add support for Flink Application Version

```
make test-acc PKG=flink 
TF_ACC=1 CGO_ENABLED=0 go test ./internal/service/flink/... \
-v -count 1 -parallel 10   -timeout 180m
2023/02/09 23:27:42 [DEBUG] Creating an instance of kafkaTopicCache ...
=== RUN   TestAccAivenFlinkApplicationVersion_basic
=== PAUSE TestAccAivenFlinkApplicationVersion_basic
=== RUN   TestAccAiven_flink
=== PAUSE TestAccAiven_flink
=== CONT  TestAccAivenFlinkApplicationVersion_basic
=== CONT  TestAccAiven_flink
--- PASS: TestAccAiven_flink (302.89s)
--- PASS: TestAccAivenFlinkApplicationVersion_basic (357.26s)
PASS
ok      github.com/aiven/terraform-provider-aiven/internal/service/flink        357.267s
```

Need to add docs otherwise seem to be working

### CODE REVIEW

File: internal/schemautil/userconfig/handle.go [hunk 1]
```diff
@@ -180,7 +180,7 @@ func handleArrayProperty(n string, p map[string]interface{}, t string) (map[stri
 
 		if an == "ip_filter" {
 			// TODO: Add ip_filter_object to this sanity check when DiffSuppressFunc is implemented for it.
-			s[jen.Id("DiffSuppressFunc")] = jen.Qual(SchemaUtilPackage, "IpFilterArrayDiffSuppressFunc")
+			s[jen.Id("DiffSuppressFunc")] = jen.Qual(SchemaUtilPackage, "IPFilterArrayDiffSuppressFunc")
```
Comment (Serpentiel) [inline internal/schemautil/userconfig/handle.go:183]: not sure, but I believe we already had this changed

File: internal/service/flink/flink_application_version.go [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+	if err != nil {
+		return diag.Errorf("cannot create Flink Application Version: %+v - %v", expandFlinkApplicationVersionSourcesOrSinks(d.Get("sources").(*schema.Set).List()), err)
+	}
+
+	d.SetId(schemautil.BuildResourceID(project, serviceName, applicationID, r.ID))
+
+	return resourceFlinkApplicationVersionRead(ctx, d, m)
+}
+
+func expandFlinkApplicationVersionSourcesOrSinks(sources []interface{}) []aiven.FlinkApplicationVersionRelation {
```
Comment (Serpentiel) [inline internal/service/flink/flink_application_version.go:132]: nit: could we have all functions commented please?

**Review by Serpentiel — DISMISSED** [review]

File: internal/schemautil/userconfig/handle.go (hunk 1 above)
Comment (ivan-savciuc) [reply internal/schemautil/userconfig/handle.go:183]: This is what is happening in the master 
```
➜  terraform-provider-aiven git:(main) go generate
ok      github.com/aiven/terraform-provider-aiven/internal/schemautil/userconfig        0.073s
➜  terraform-provider-aiven git:(main) ✗ git status
On branch main
Your branch is up to date with 'origin/main'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   internal/schemautil/userconfig/dist/service_types.go

no changes added to commit (use "git add" and/or "git commit -a")
```
And diff :
```
@@ -64,9 +64,9 @@ func ServiceTypeCassandra() *schema.Schema {
                },
                "ip_filter": {
                        Description:      "Allow incoming connections from CIDR address block, e.g. '10.20.0.0/16'",
-                       DiffSuppressFunc: schemautil.IPFilterArrayDiffSuppressFunc,
+                       DiffSuppressFunc: schemautil.IpFilterArrayDiffSuppressFunc,
                        Elem: &schema.Schema{
-                               DiffSuppressFunc: schemautil.IPFilterValueDiffSuppressFunc,
+                               DiffSuppressFunc: schemautil.IpFilterValueDiffSuppressFunc,
                                Type:             schema.TypeString,
                        },
```

Comment (ivan-savciuc) [reply internal/schemautil/userconfig/handle.go:183]: So I guess we have to fix that

File: internal/service/flink/flink_application_version.go (hunk 2 above)
Comment (ivan-savciuc) [reply internal/service/flink/flink_application_version.go:133]: ok, no problem added more comments even to the private functions

**Comment by byashimov** [comment]
LGTM

**Review by byashimov — APPROVED** [review]

File: internal/schemautil/userconfig/handle.go (hunk 1 above)
Comment (Serpentiel) [reply internal/schemautil/userconfig/handle.go:183]: ahh sorry, this is fixed in v4 branch

**Pull request merged**